lib/SymbolGraphGen/DeclarationFragmentPrinter.cpp

```diff
@@ -189,9 +189,17 @@ void DeclarationFragmentPrinter::printAbridgedType(const GenericTypeDecl *TD,
       case DeclKind::Protocol:
         printText(getTokenText(tok::kw_protocol));
         break;
-      case DeclKind::Class:
-        printText(getTokenText(tok::kw_class));
-        break;
+        case DeclKind::Class: {
+            auto *newthing = cast<ClassDecl>(TD);
+            if (newthing->isActor()) {
+                printText("actor");
+            } else if (newthing->isDistributedActor()) {
+                printText("distributed actor");
```

Comment:
Are we sure that distributed actors won't fall into the first branch? I'm unfamiliar with the underlying decl types so i dunno if this will cause problems.

Reply:
I'm not sure, but I'll definitely do my best to add tests for both an "actor" and "distributed actor", as well as re-read through the declaration types with the detail that Slava provided in the forums.

Reply:
These are semantic requests here - are you sure you want to go all the way through to protocol resolution to determine if something is declared as a `class` or an `actor` or a `distributed actor`? Since you know it's a `ClassDecl`, you can use `newthing->isExplicitActor()` and `newthing->isExplicitDistributedActor()` to do a syntactic check.

Reply:
The semantic request `isActor()` will, for example, see this as an actor

```
class MyCustomActor: Actor { /**/ }
```

and this as a distributed actor

```
class MyDistributedActor: DistributedActor { /**/ }
```

Reply:
Ah, thank you @CodaFi - I'll switch that up, sounds like I touched on the wrong API here for what I was after.